conjure-go-client/httpclient/response_error_decoder_middleware.go

```diff
@@ -73,3 +75,28 @@ func (d restErrorDecoder) DecodeError(resp *http.Response) error {
 func StatusCodeFromError(err error) (statusCode int, ok bool) {
 	return internal.StatusCodeFromError(err)
 }
+
+func ConjureErrorDecoder() ErrorDecoder {
```

Comment:
This is currently opt-in, but we probably just want to make it the default.

Reply:
I'd argue for just making this the default. If we wanted to avoid breaking backcompat (except where we have real typed errors), we could make the fallback error message the same as the current restErrorDecoder one, but don't think that's really necessary.